app/workers/sidekiq_batch_id_cleaner.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class SidekiqBatchIdCleaner < ApplicationJob
+
+  KEEP_TTL = 3.days
+
+  def perform(bid)
+    # TODO: ensure there is no task running still
+    batch_key = "BID-#{bid}"
+    now = Time.zone.now
+    timestamp = System.redis.hget(batch_key, 'created_at')
+    created_at = timestamp ? Time.zone.at(timestamp.to_f) : now
+
+    # Keep batches for 3 days
+    return if created_at > now - KEEP_TTL
+
+    Sidekiq::Batch.cleanup_redis(bid)
+  end
+
+  class Enqueuer < ApplicationJob
+
+    MAX_FETCH_COUNT = 1000
+
+    def perform(*)
+      System.redis.scan_each match: 'BID-*', count: MAX_FETCH_COUNT do |key|
+        next if key =~ /-(success|complete|failed|jids)$/
```

Comment:
we do not clean-up jobs with these statuses?

Reply:
It is a bugf in sidekiq-batch ...

Reply:
These are filtered out, because we only need the ID here (so, we're taking the "parent" key, e.g. `BID-123456789`), all of the keys with suffixes are going to be removed by `Sidekiq::Batch.cleanup_redis(bid)`, see the [source](https://github.com/breamware/sidekiq-batch/blob/e1cdc037fba01995dae112ebd94ae71bb2e1d1cd/lib/sidekiq/batch.rb#L265-L279)